src/lib/list/list.ts

```diff
@@ -146,9 +158,9 @@ export class MdListItem extends _MdListItemMixinBase implements AfterContentInit
   constructor(private _renderer: Renderer2,
               private _element: ElementRef,
               @Optional() private _list: MdList,
-              @Optional() navList: MdNavListCssMatStyler) {
+              @Optional() private _navList: MdNavList) {
```

Comment:
This should potentially have `@Host` as well, otherwise it'll look up the component tree.